### ISSUE TITLE
Reimplementation of a graph-based recommender system SR-GNN?

### ISSUE THREAD

**Issue by RuihongQiu** [rusty1s/pytorch_geometric#200]
## ❓ Questions & Help

I have used PyTorch_Geometric to reimplement SR-GNN in this [repo](https://github.com/RuihongQiu/SR-GNN_PyTorch-Geometric) but still come across some problems in training to converge.

Anyone interested in reimplement the [Session-based Recommendation with Graph Neural Networks](https://arxiv.org/pdf/1811.00855.pdf)?

**Comment by RuihongQiu** [comment]
I have finished the reimplementation in my [SR-GNN_PyTorch-Geometric](https://github.com/RuihongQiu/SR-GNN_PyTorch-Geometric) repo and achieve the similar result with the paper.

Welcome pull request!

**Comment by rusty1s** [comment]
Really cool :) Would love to see a pull request on this one if you have time.

**Comment by RuihongQiu** [comment]
But the full code includes a folder and it seems too heavy to add into the `examples` folder in your code.

**Comment by rusty1s** [comment]
I do not think there is a need to include the raw dataset files into PyG. Just download them on the fly like we do it with all datasets.

**Comment by RuihongQiu** [comment]
Yeah. I have 4 files. And do I need to create a folder inside `examples` to put them in?

**Comment by rusty1s** [comment]
I suggest adding the datasets to the `torch_geometric.datasets` subpackage, and the model to the `torch_geometric.nn.models` subpackage. You can then create a single-file example to train the model (*e.g.*, like `infomax.py` or `autoencoder.py`). What do you think?

**Comment by RuihongQiu** [comment]
Hmm. Good idea and I will have a try : )

**Comment by Jeriousman** [comment]
Did you make it happen @RuihongQiu? If so, where can I find your work?
